README.md

```diff
@@ -1,32 +1,86 @@
+[![PyPI - Python Version](https://img.shields.io/pypi/pyversions/ragbits)](https://pypi.org/project/ragbits)
 
-## Build from source
+## What's Included?
 
-To build and run Ragbits from the source code:
+- [X] **[Core](packages/ragbits-core)** - Fundamental tools for working with prompts and LLMs.
+- [X] **[Document Search](packages/ragbits-document-search)** - Handles vector search to retrieve relevant documents.
+- [X] **[CLI](packages/ragbits-cli)** - The `ragbits` shell command, enabling tools such as GUI prompt management.
+- [ ] **Flow Controls** - Manages multi-stage chat flows for performing advanced actions (coming soon).
+- [ ] **Structured Querying** - Queries structured data sources in a predictable manner (coming soon).
+- [ ] **Caching** - Adds a caching layer to reduce costs and response times (coming soon).
+- [ ] **Observability & Audit** - Tracks user queries and events for easier troubleshooting (coming soon).
+- [ ] **Guardrails** - Ensures response safety and relevance (coming soon).
 
-1. Requirements: [**uv**](https://docs.astral.sh/uv/getting-started/installation/) & [**python**](https://docs.astral.sh/uv/guides/install-python/) 3.10 or higher
-2. Install dependencies and run venv in editable mode:
+## Installation
 
-```bash
-$ source ./setup_dev_env.sh
+To use the complete Ragbits stack, install the `ragbits` package:
+
+```sh
+pip install ragbits
+```
+
+Alternatively, you can use individual components of the stack by installing their respective packages: `ragbits-core`, `ragbits-document-search`, `ragbits-cli`.
+
+## Quickstart
+
+First, create a prompt and a model for the data used in the prompt:
+
+```python
+from pydantic import BaseModel
+from ragbits.core.prompt import Prompt
+
+class Dog(BaseModel):
+    breed: str
+    age: int
+    temperament: str
+
+class DogNamePrompt(Prompt[Dog, str]):
+    system_prompt = """
+    You are a dog name generator. You come up with funny names for dogs given the dog details.
+    """
+
+    user_prompt = """
+    The dog is a {breed} breed, {age} years old, and has a {temperament} temperament.
+    """
 ```
 
-## Install pre-commit
+Next, create an instance of the LLM and the prompt:
 
-To ensure code quality we use pre-commit hook with several checks. Setup it by:
+```python
+from ragbits.core.llms.litellm import LiteLLM
 
+llm = LiteLLM("gpt-4o")
+example_dog = Dog(breed="Golden Retriever", age=3, temperament="friendly")
+prompt = DogNamePrompt(example_dog)
 ```
-pre-commit install
+
+Finally, generate a response from the LLM using the prompt:
+
+```python
+response = await llm.generate(prompt)
+print(f"Generated dog name: {response}")
 ```
 
-All updated files will be reformatted and linted before the commit.
+<!--
+TODO:
+Add links to quickstart guides for individual packages, demonstrating how to extend this with their functionality.
+Add a link to the full tutorial.
+-->
+
+## Documentation
```

Comment:
I'd remove this section and simply add some links at the top of the README to the company's website (for now, until we have an lp for Ragbits), documentation and email contact. I'm thinking of something like this:

<h3>
    
[Homepage](https://deepsense.ai) | [Documentation](https://ragbits.deepsense.ai/) | [Contact](mailto:ragbits@deepsense.ai)

</h3>

More detailed example from db-ally README -> https://github.com/deepsense-ai/db-ally/blob/mp/update-readme/README.md

Reply:
ok

Reply:
Done, I hope you don't mind that I used a standard font size - that's how I usually see this done in other repositories. Marking this element as a header doesn't make much sense in my mind.